spec/unit/lwrp_spec.rb

```diff
@@ -229,14 +229,6 @@ def get_lwrp_provider(name)
         expect(klass.resource_name).to eq(:foo)
       end
 
-      context "when creating a new instance" do
-        it "raises an exception if resource_name is nil" do
-          expect {
-            klass.new('blah')
-          }.to raise_error(Chef::Exceptions::InvalidResourceSpecification)
-        end
-      end
```

Comment:
Because LWRPs are now using the same code as Resources, and resources don't worry about this.  I'm going to add this back in at the Resource level, though.